R/run_stress_test.R

```diff
@@ -1,55 +1,126 @@
-#' @param terminal_value Numeric. A ratio to determine the share of the
+#' @param terminal_value_arg Numeric. A ratio to determine the share of the
 #'   discounted value used in the terminal value calculation beyond the
 #'   projected time frame. For accepted range compare `terminal_value_range_lookup`.
-#' @param risk_free_rate Numeric that indicates the risk free rate of interest.
+#' @param risk_free_rate_arg Numeric that indicates the risk free rate of interest.
 #'   For accepted range compare `risk_free_rate_range_lookup`.
-#' @param discount_rate Numeric, that holds the discount rate of dividends per
+#' @param discount_rate_arg Numeric, that holds the discount rate of dividends per
 #'   year in the DCF. For accepted range compare `discount_rate_range_lookup`.
-#' @param div_netprofit_prop_coef Numeric. A coefficient that determines how
+#' @param div_netprofit_prop_coef_arg Numeric. A coefficient that determines how
 #'   strongly the future dividends propagate to the company value. For accepted
 #'   range compare `div_netprofit_prop_coef_range_lookup`.
-#' @param shock_year Numeric, holding year the shock is applied. For accepted
+#' @param shock_year_arg Numeric, holding year the shock is applied. For accepted
 #'   range compare `shock_year_range_lookup`.
-#' @param term Numeric. A coefficient that determines for which maturity the
+#' @param term_arg Numeric. A coefficient that determines for which maturity the
 #'   expected loss should be calculated in the credit risk section. For accepted
 #'   range compare `term_range_lookup`.
-#' @param company_exclusion Boolean, indicating if companies provided in dataset
+#' @param company_exclusion_arg Boolean, indicating if companies provided in dataset
 #'   excluded_companies.csv shall be excluded.
 #' @return NULL
 #' @export
-run_stress_test <- function(asset_type,
-                            lgd_senior_claims = 0.45,
-                            lgd_subordinated_claims = 0.75,
-                            terminal_value = 0,
-                            risk_free_rate = 0.02,
-                            discount_rate = 0.02,
-                            div_netprofit_prop_coef = 1,
-                            shock_year = 2030,
-                            term = 2,
-                            company_exclusion = TRUE) {
-
-  cat("-- Validating input arguments. \n")
+run_stress_test <- function(asset_type_arg,
+                            lgd_senior_claims_arg = 0.45,
+                            lgd_subordinated_claims_arg = 0.75,
+                            terminal_value_arg = 0,
+                            risk_free_rate_arg = 0.02,
+                            discount_rate_arg = 0.02,
+                            div_netprofit_prop_coef_arg = 1,
+                            shock_year_arg = 2030,
+                            term_arg = 2,
+                            company_exclusion_arg = TRUE) {
+  cat("Running transition risk stress test \n")
```

Comment:
Why not use `message()`? It seems to be the right tool for this job.

Reply:
messages cannot be formatted as nicely (in the sense of conssitent intentation etc.) as cats for log output in the console I think. For now we communicate with the user via cat and warning, I would rather keep this consistent until we maybe decide to switch to another logging approach

Reply:
OK, I'm confident we'll find a way to give messages with a better tool and with the formatting you want. Just for the record:

>  The purposes of `cat()` and `message()` are different. Use `cat()` when the primary role of the function is to print to the console, like `print()` or `str()` methods. Use `message()` as a side-channel to print to the console when the primary purpose of the function is something else. In other words, `cat()` is for when the user asks for something to be printed and `message()` is for when the developer elects to print something.

https://adv-r.hadley.nz/conditions.html